Refresh the Backup and Restore guide

ptgott opened the pull request:
Closes #31866
Closes #14537
Closes #51782
Closes #30843 

- Recommend IaC for dynamic resources instead of using `tctl`.
- Add a separate section on using `tctl get all` on the Auth Service machine.
- Remove tabs in service state section to simplify the guide.
- Condense the state type table. Condense different rows re: dynamic resources into a single row to make the table shorter. Also make this a generic table, rather than an example table for DynamoDB and S3 users.
- Assume the guide is about the Auth Service. Adjust sections accordingly.
- Add an item to the backend table re: Azure Postgres.

ptgott committed:
Refresh the Backup and Restore guide

Closes #31866
Closes #14537
Closes #51782

- Recommend IaC for dynamic resources instead of using `tctl`.
- Add a separate section on using `tctl get all` on the Auth Service
  machine.
- Remove tabs in service state section to simplify the guide.
- Condense the state type table. Condense different rows re: dynamic
  resources into a single row to make the table shorter. Also make this
  a generic table, rather than an example table for DynamoDB and S3
  users.
- Assume the guide is about the Auth Service. Adjust sections
  accordingly.
- Add an item to the backend table re: Azure Postgres. This closes
  #30843.

mmcallister commented on docs/pages/admin-guides/management/operations/backup-restore.mdx:
```diff
@@ -2,211 +2,114 @@
-its backend initialization on first startup, so it is safe for use in
-supervised/High Availability contexts.
-
-### Limitations
```
I know we're moving away from using the limitations heading but this seems like important info to call out in some way; would 'Considerations' be better?

ptgott replied:
Changed in 8ab80237710d1e336f6dda07a2f856a7077dab44

zmb3 commented on docs/pages/admin-guides/management/operations/backup-restore.mdx:
```diff
@@ -2,211 +2,118 @@
+
+## Backing up Teleport backends
+
+Your plan for backing up the Teleport cluster state and session recording
```
The table below only covers the cluster state backends - it doesn't mention session recording backends (S3-compatible storage) at all.

ptgott replied:
Adding a separate table for session recording backends

zmb3 commented on docs/pages/admin-guides/management/operations/backup-restore.mdx:
```diff
@@ -2,211 +2,118 @@
+Teleport uses [dynamic
+resources](../../infrastructure-as-code/infrastructure-as-code.mdx) for roles,
+local users, authentication connectors, and other configurations. We recommend
+that you back up your configuration resources because they are essential to the
+normal operation of your cluster.
```
But this information is backed up if you back up the cluster state backend. Should we make that more clear?

zmb3 commented on docs/pages/admin-guides/management/operations/backup-restore.mdx:
````diff
@@ -2,211 +2,118 @@
-# Export dynamic configuration state from old cluster
-$ tctl get all --with-secrets > state.yaml
+   ```code
+   $ sudo teleport start --config fresh.yaml --bootstrap state.yaml
````
This might confuse people who use one of our systemd-based package installs.

Are we suggesting that users
- stop the systemd service
- start teleport manually with `--bootstrap` (not using systemd)
- stop the manually started version and restart the systemd service
?

ptgott replied:
👍 I've removed the `--bootstrap` discussion and added a section about `teleport backend clone`.

zmb3 commented on docs/pages/admin-guides/management/operations/backup-restore.mdx:
```diff
@@ -219,23 +126,7 @@ also apply to a new cluster being bootstrapped from the state of an old cluster:
-
-</TabItem>
-</Tabs>
+  - Any Teleport Agent whose invite token is defined in the Auth Service's
```
Is this true? 

Agents that already have host certificates shouldn't need a join token, right?

ptgott committed:
Respond to feedback on Backup and Restore edits

- Add a session recording backend table to complement the cluster state
  backend table.
- Edit the intro paragraph of the IaC section to talk about its
  relationship to cluster state backend backups.
- Remove the "Using a state file" section. The instructions in this
  section are difficult to implement for package manager installations
  of Teleport, which use a `systemd` service.
- Add Cloud SQL to the cluster state backend table.
- Add a section about `teleport backend clone`.

ptgott commented on docs/pages/admin-guides/management/operations/backup-restore.mdx:
```diff
@@ -2,211 +2,118 @@
 | DynamoDB | [Follow AWS's guidelines for backup and restore](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/BackupRestore.html) |
 | etcd | [Follow etcd's guidelines for disaster recovery](https://etcd.io/docs/v2/admin_guide) |
 | Firestore | [Follow GCP's guidelines for automated backups](https://firebase.google.com/docs/database/backups) |
+| Azure Database for PostgreSQL | [Follow Azure's guidelines for automated backups](https://learn.microsoft.com/en-us/azure/backup/backup-azure-database-postgresql) |
```
I'll add this

ptgott commented on docs/pages/admin-guides/management/operations/backup-restore.mdx:
```diff
@@ -2,211 +2,118 @@
-  - Any Node whose invite token is defined in the Auth Service's configuration
-    file will be able to join automatically, but Nodes that were added
-    dynamically will need to be re-invited.
+#### Restarting the Auth Service with a state file
```
I'll add a section about `teleport backend clone`

ptgott committed:
Respond to rosstimothy feedback

- Specify the audit event backend for audit events in the first table.
- Swap src and dst in the clone example.
- Mention the audit event backend separately from the cluster state
  backend.
- Remove the mention of S3 compatibility when describing the recording
  backend.
- Edit the `backend clone` docs. Use "items" instead of "dynamic
  resources". Don't instruct the user to stop Auth Service instances.
